src/unix/config/discover.ml

```diff
@@ -705,6 +705,44 @@ struct
       |}
   }
 
+  let () = feature {
+    pretty_name = "madvise MERGEABLE";
+    macro_name = "HAVE_MADVISE_MERGEABLE";
+    detect = fun context ->
+      skip_if_windows context @@ fun () ->
+      compiles context {|
+        #define _GNU_SOURCE
+        #include <sys/mmap.h>
+        int main()
+        {
+        #if !defined(MADV_MERGEABLE) && !defined(MADV_UNMERGEABLE)
+        #error \"not available\"
+        #else
+            return 0;
+        #endif
+        }
+      |}
+  }
+
+  let () = feature {
+    pretty_name = "madvise HUGEPAGE";
+    macro_name = "HAVE_MADVISE_HUGEPAGE";
+    detect = fun context ->
+      skip_if_windows context @@ fun () ->
+      compiles context {|
+        #define _GNU_SOURCE
+        #include <sys/mmap.h>
+        int main()
+        {
+        #if !defined(MADV_HUGEPAGE) && !defined(MADV_NOHUGEPAGE)
+        #error \"not available\"
+        #else
+            return 0;
+        #endif
+        }
+      |}
+  }
```

Comment:
Given you don't use `HAVE_MADVISE_HUGEPAGE` or `HAVE_MADVISE_MERGEABLE` in the `#fidef`s in C, you probably don't need to detect them in `discover.ml` at all.

If you do decide that you need detection in `discover.ml`, I'd ask that, since the code between these is almost the same, you make one function for all the flags, and call it once for each flag to detect. See `nanosecond_stat` detection code below.

Reply:
I will remove detection then, it's not really usable outside `madvise()` calls.